### PR TITLE
coverage: use lcov file format

### PR DESCRIPTION
**Pull request by vtjnash**
@staticfloat is there any way to try this out before merging?

### CODE REVIEW

**Comment by vtjnash** [comment]
CI says it doesn't like this https://github.com/JuliaCI/julia-buildbot/blob/0c28d8aa5b0d1cb7e99d60757cb59c7456f0d1d4/master/inventory.py#L176
```
builtins.TypeError: unsupported operand type(s) for +: 'map' and 'map'
```

**Comment by staticfloat** [comment]
That error was fixed by upgrading to python 3 in 71ed497, rebase and try again.

File: master/coverage.py [hunk 1]
```diff
@@ -116,16 +107,21 @@
...
-        command=[util.Interpolate("%(prop:juliadir)s/bin/julia"), "--sysimage-native-code=no", "--code-coverage=all", "--inline=no", "-e", run_coverage_cmd],
-        timeout=7200,
-    ),
+    #steps.ShellCommand(
+    #    name="Run non-inlined tests",
+    #    command=[util.Interpolate("%(prop:juliadir)s/bin/julia"),
+    #             "--sysimage-native-code=no", util.Interpolate("--code-coverage=%(prop:juliadir)s/LCOV/cov-%%p.info"), "--inline=no",
+    #             "-e", run_coverage_cmd],
+    #    timeout=7200,
+    #),
```
Comment (staticfloat) [inline master/coverage.py]: Do we no longer have non-inlined tests?

Comment (vtjnash) [reply master/coverage.py]: We haven't in a couple years, although mainly I was just selfishly interested in getting results from the buildbot faster.

Comment (staticfloat) [reply master/coverage.py]: Sounds good to me, I'll just remove this chunk completely then after merging.

**Pull request merged**